src/utils/automation-helper.js

```diff
@@ -29,10 +29,337 @@ export const getAutomation = async (automationId) => {
+  // Max sleep time ~10 minutes
+  const sleepTimes = [
+    2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 15, 15, 15, 15, 15, 15,
+    30, 30, 30, 30, 60, 60, 60, 60, 60, 60,
+  ];
+  let finalRun;
+  for (let i = 0; i < sleepTimes.length; i++) {
+    // eslint-disable-next-line no-await-in-loop
+    const automationRun = await getAutomationRun(automationId, runId);
+    const { status } = automationRun;
+    if (!runningStatuses.includes(status)) {
+      finalRun = automationRun;
+      break;
+    } else {
+      // eslint-disable-next-line no-await-in-loop
+      await sleep(sleepTimes[i] * 1000);
+    }
+  }
+  let msg;
+  switch (finalRun.status) {
+    case 'finished': {
+      if (finalRun.title?.length > 0) {
+        msg = parseOutput(finalRun.title, translator);
+        msg.ok = true;
+      } else {
+        msg = { message: translator.get('Object.ActionButton.Automation.DefaultAutomationMsg'), ok: true };
+      }
+      break;
+    }
+    case 'failed': {
+      if (finalRun.title?.length > 0) {
+        msg = parseOutput(finalRun.title, translator);
+        msg.ok = false;
+      } else {
+        msg = { message: translator.get('Object.ActionButton.Automation.AutomationError'), ok: true };
+      }
+      break;
+    }
+    case 'finished with warnings': {
+      if (finalRun.title?.length > 0) {
+        msg = parseOutput(finalRun.title, translator);
+        msg.ok = false;
+      } else {
+        msg = { message: translator.get('Object.ActionButton.Automation.DefaultAutomationMsg'), ok: true };
+      }
+      break;
+    }
+    case 'must stop':
+    case 'stopped': {
+      if (finalRun.title?.length > 0) {
+        msg = parseOutput(finalRun.title, translator);
+        msg.ok = false;
+      } else {
+        msg = { message: translator.get('Object.ActionButton.Automation.UnkownError'), ok: true };
+      }
+      break;
+    }
+    default: {
+      if (finalRun.title?.length > 0) {
+        msg = parseOutput(finalRun.title, translator);
+        msg.ok = true;
+      } else {
+        msg = { message: translator.get('Object.ActionButton.Automation.DefaultAutomationMsg'), ok: true };
+      }
+    }
+  }
+  return msg;
+};
+
+export const getAutomationMsg = async (automationId, triggered, response, translator) => {
+  let message;
+  switch (response.status) {
+    case 200:
+    case 201: {
+      const data = await response.json();
+      const { status, guid, id } = data;
+      const queued = status === 'queued';
+      const runId = typeof id === 'undefined' ? guid : id;
+      if (!triggered || queued) {
+        message = automationRunPolling(automationId, runId, translator);
+      } else {
+        message = parseOutput(data, translator);
+        message.ok = true;
+      }
+      break;
+    }
+    case 400: {
+      message = { message: translator.get('Object.ActionButton.Automation.BadRequest'), ok: false };
```

Comment:
Very good point :) Since all of the translations that are needed here are error messages, I solved this by defaulting to the English string like this: `translator.get('Object.ActionButton.Automation.DefaultAutomationMsg') || 'Automation finished'`

Reply:
That won't do. The translator will not return something in a false state, but instead it returns the input string

Reply:
Ah you are right - I updated this in the latest commit

Reply:
Well it is not really fixed I would say. Also the error messages need to be translated. If you want I can help with the setup.